corehq/apps/hqwebapp/static/hqwebapp/js/main.js

```diff
@@ -399,6 +401,34 @@ hqDefine('hqwebapp/js/main', [
             }
         }
 
+        function isUsingRequireJS() {
+            return (typeof define === 'function' && define.amd && window.USE_REQUIREJS);
+        }
+
+        function unsupportedBrowser() {
+            // check explicitly for Safari. Relying on browser capabilities would be preferred,
+            // but our issue with Safari is described here: https://dimagi-dev.atlassian.net/browse/SUPPORT-4778
+            // (history.replaceState raises security exceptions that aren't present in other browsers).
+            // This can be verified here: (https://jsfiddle.net/j1sxxLwy/),
+            // but it's not something that can be efficiently feature-checked
+            if (window.safari !== undefined) {
+                return true;    // found a Safari browser
+            }
+
+            modernizr = isUsingRequireJS() ? modernizr : window.Modernizr;
```

Comment:
Haven't read through the whole PR yet, but have you seen [this code](https://github.com/dimagi/commcare-hq/blob/08640a777f11c28813cf242b7417b1e850b59c4c/corehq/apps/hqwebapp/static/hqwebapp/js/hqModules.js#L58-L67)? This is how other third-party code that relies on `window` pre-RequireJS is set up in HQ.

Reply:
Thanks for pointing this out! Modified to use the version you highlighted